src/main/java/com/epam/training/gen/ai/service/GenAiChatService.java

```diff
@@ -0,0 +1,80 @@
+package com.epam.training.gen.ai.service;
+
+import com.azure.ai.openai.models.CompletionsUsage;
+import com.epam.training.gen.ai.model.ChatInput;
+import com.epam.training.gen.ai.model.ChatMessage;
+import com.epam.training.gen.ai.model.ChatOutput;
+import com.microsoft.semantickernel.Kernel;
+import com.microsoft.semantickernel.orchestration.InvocationContext;
+import com.microsoft.semantickernel.orchestration.PromptExecutionSettings;
+import com.microsoft.semantickernel.services.chatcompletion.AuthorRole;
+import com.microsoft.semantickernel.services.chatcompletion.ChatCompletionService;
+import com.microsoft.semantickernel.services.chatcompletion.ChatHistory;
+import com.microsoft.semantickernel.services.chatcompletion.ChatMessageContent;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class GenAiChatService {
+
+    private final ChatCompletionService chatCompletionService;
```

Comment:
it's not necessary to pass ChatCompletionService here explicitly. you could take it from kernel as here: 

ChatCompletionService chatCompletionService = kernel.getService(ChatCompletionService.class);

https://learn.microsoft.com/en-us/semantic-kernel/concepts/ai-services/chat-completion/?tabs=csharp-AzureOpenAI%2Cpython-AzureOpenAI%2Cjava-AzureOpenAI&pivots=programming-language-java